services/blog.py

```
@@ -389,3 +394,172 @@ def _generate_content(self, digest: Dict[str, Any]) -> str:
+            raise ValueError(f"target_date must be YYYY-MM-DD format, got: {target_date}")
+        
+        # Check if digest exists in digests directory first
+        digests_dir = Path("digests")
+        digest_path = digests_dir / f"{target_date}.json"
+        
+        # Ensure the resolved path stays within the digests directory
+        try:
+            digests_dir_resolved = digests_dir.resolve()
+            resolved_path = digest_path.resolve()
+            resolved_path.relative_to(digests_dir_resolved)
+        except (OSError, RuntimeError, ValueError) as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)